### DISCUSSION TITLE
Does clippy provide symbol analysis (via API or tools)?

### DISCUSSION THREAD

**Discussion by JamesBear** [rust-lang/rust-clippy#8682] [Q&A] [answered]
Clippy provides useful suggestions to write better statements, all kinds of statements. So I guess Clippy knows what symbols are in a statement and which crates the symbols are from? This alone can provide useful information in helping analyzing symbol statistics (i.e. which functions are called frequently) and lay out crate dependencies. 

The question is, can I get result of symbol analysis via Clippy?

**Comment by flip1995** [comment, marked as answer]
Not via Clippy. But you can write your own compiler driver to do that. Have a look at the [`RunCompiler`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_driver/struct.RunCompiler.html) interface. You can register callbacks, like a list of lints that should run. in those lints you can collect that information and then output it at the end.
